From a0818e067f4503ad0a000585aeffdbb52316bbb0 Mon Sep 17 00:00:00 2001
From: Richard Tang <richard.tang@intel.com>
Date: Thu, 2 May 2013 15:55:15 +0800
Subject: libstagefright: remove 10ms polling in AwesomePlayer

BZ: 34320

AwesomePlayer will wake up every 10ms while video is playing.

This patch will remove the 10ms polling.

And it will save 18mW according to Melwyn's measurement data.

Category: aosp improvement
Domain: Video.Media-local playback
Origin: internal
Upstream-Candidate: yes

Change-Id: I77bf1d5b4708ec0340475b888aeb2245dbecd72e
Orig-Change-Id: I0ee1079fb3565bc94cc1bee4ed4cae5d5b527a8a
Signed-off-by: Richard Tang <richard.tang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index 4241dae..12c8a84 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -1950,9 +1950,13 @@ void AwesomePlayer::onVideoEvent() {
             }
         }
 
-        if (latenessUs < -10000) {
+        if (latenessUs < -30000) {
+            // We're more than 30ms early.
+            postVideoEvent_l(30000);
+            return;
+        } else if (latenessUs < -10000) {
             // We're more than 10ms early.
-            postVideoEvent_l(10000);
+            postVideoEvent_l(-latenessUs);
             return;
         }
     }
-- 
1.7.4.1

